dxr/static_unhashed/js/context_menu.js

```diff
@@ -6,7 +6,16 @@ $(function() {
     // Get the file content container
     var fileContainer = $('#file'),
         queryField = $('#query'),
-        contentContainer = $('#content');
+        contentContainer = $('#content'),
+        nonWordCharRE = /[^A-Z0-9_]/i;
+
+    if (['cpp', 'cc', 'cxx', 'h', 'hxx', 'hpp'].indexOf(
+          extension(window.location.pathname).toLowerCase()) > -1) {
```

Comment:
I'm not wild about doing this, so if people have better ideas feel free to let me know :-)
The only issue I can think of with just always including '~' as a word character is that in JS you see `~~` sometimes, and since we don't have analysis to separate the `~~` from what it's acting on with markup, the JS will consider `~~thing` to be a word it should search on, which isn't really what we want.